src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_TypeChar.cs

```diff
@@ -357,12 +373,12 @@ private bool IsFilterCharacter(char ch)
             return GetCompletionRules().IsFilterCharacter(selectedItem, ch, filterText);
         }
 
-        private string GetCurrentFilterText(Model model, CompletionItem selectedItem)
+        private string GetCurrentFilterText(Model model, CompletionItem selectedItem, bool stopBeforeCaret = true)
```

Comment:
`GetCurrentFilterText` does almost exactly the same thing as your code above that tracks the edit span forward in the view--consider turning this into `GetCUrrentFilterTextAndViewSpan` and returning the view span as an out param?